### ISSUE TITLE
Namespace System.Text.Json.Node may conflict with variable names in VB.NET projects when using default imports

### ISSUE THREAD

**Issue by pranavkm** [dotnet/runtime#51637]
This popped up as part of some app building. Repro:

```xml
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <RootNamespace>ClassLibrary2</RootNamespace>
    <TargetFramework>net6.0</TargetFramework>
  </PropertyGroup>

  <ItemGroup>
    <Import Include="System.Text.Json" />
  </ItemGroup>

</Project>
```

```vb
Public Class Class1
    Public Function Foo()
        For Each node In Enumerable.Range(0, 10)
            node.ToString()
        Next
        Return 0
    End Function
End Class
```

Using a preview5 M.NETCore.App targeting pack, this results in a compilation error:

```
ClassLibrary2\Class1.vb(4,18): error BC30112: 'System.Text.Json.Node' is a namespace and cannot be used as an expression.
ClassLibrary2\Class1.vb(5,13): error BC30456: 'ToString' is not a member of 'System.Text.Json.Node'.
```

Renaming the variable to a differently named value (e..g `For Each x`) solves the issue.

**Comment by ericstj** [comment]
@terrajobst @bartonjs - has this come up for API review before?  Do we have a stance for namespace collisions? 

@cston @KathleenDollard - is it correct for VB to prefer a relative namespace over a local identifier?  That seems pretty unusual.  Is there a way to tell VB not to do that in our API (like an attribute or something).

**Comment by KathleenDollard** [comment]
@AlekseyTs can you confirm that this is not surprising behavior, and perhaps comment on the issue.

**Comment by AlekseyTs** [comment]
The local isn't explicitly declared anywhere, if the name wouldn't bind to anything, then it would be considered as an implicitly declared local. There is an option to use an explicit declaration, by adding ```As <Type>``` to the identifier that is supposed to declare a local.

**Comment by PriyaPurkayastha** [comment]
Is there a specific change recently that has surfaced this issue? My AppCompat team confirmed that when they had built this app with .NET 6 Preview 1 and Preview 2 bits, they didn't encounter this failure.
cc @jiangzeng01

**Comment by AlekseyTs** [comment]
I don't believe we changed anything in VB compiler recently. It is possible that the "Node" namespace was added in the framework

**Comment by terrajobst** [comment]
I think we should avoid the name `Node` as a namespace TBH. While VB's binding is a bit odd it's also somewhat confusing as a namespace name. Namespaces shouldn't use names that could be type names. For example, a `Syntax` isn't likely something would use as a type name, but `SyntaxNode` looks like a type name. Consider that fully qualified names often appear in output, such as the VB error message, and I was looking for a type name, not a namespace name.

Normally, we use either use nouns for namespace that aren't likely types (e.g. `IO`) or pluralized names (e.g. `Collections`). So I'd either go for `Nodes` or something unrelated (e.g. `Linq`).

/cc @bartonjs

**Comment by bartonjs** [comment]
Yeah, I agree with @terrajobst.  `Nodes`, `DOM`, `ObjectModel`, `Linq`, etc.

**Comment by ericstj** [comment]
cc @steveharter Was the namespace something that was a focus of API review already?

**Comment by steveharter** [comment]
Using a "Nodes" was not discussed, but that makes the most sense to me.

We didn't want to place it in the "S.T.J" namespace directly as it would be confusing to have it there with `JsonDocument` and `JsonElement` which are not directly related but sound the same functionality-wise.

Using `Linq` means nothing to most users of the feature; Newtonsoft has that naming FWIW.

**Comment by ericstj** [comment]
Fixed in #51937
